src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -749,17 +752,26 @@ public static ushort TrailingZeroCount(ushort value)
         /// <summary>
         /// <see cref="ByteToNormalizedFloat"/> as many elements as possible, slicing them down (keeping the remainder).
         /// </summary>
+        /// <param name="source">The source buffer.</param>
+        /// <param name="destination">The destination buffer.</param>
         [MethodImpl(InliningOptions.ShortMethod)]
         internal static void ByteToNormalizedFloatReduce(
             ref ReadOnlySpan<byte> source,
-            ref Span<float> dest)
+            ref Span<float> destination)
         {
-            DebugGuard.IsTrue(source.Length == dest.Length, nameof(source), "Input spans must be of same length!");
+            DebugGuard.IsTrue(source.Length == destination.Length, nameof(source), "Input spans must be of same length!");
 
-            if (Avx2.IsSupported || Sse2.IsSupported)
+            if ((Vector512.IsHardwareAccelerated && Avx512F.IsSupported) ||
+                Avx2.IsSupported ||
+                Sse2.IsSupported ||
+                AdvSimd.IsSupported)
```

Comment:
```suggestion
            if (Vector128.IsHardwareAccelerated)
```

The check as written won't prevent you running the AVX-512 code on the slow processors, because those will still pass the `Avx2.IsSupported` check and enter this block, where they'll also pass the `Avx512F.IsSupported` check below and use `Vector512` anyway.

This entire expression boils down to the lowest common denominator, which is `Sse2.IsSupported || AdvSimd.IsSupported`.  That would seem to be unnecessarily restrictive though, because your `Vector128` path is 100% xplat intrinsics.  I don't know if `Vector128.IsHardwareAccelerated` reports true for wasm on .NET 8, but if it does now or in .NET 9, you'd be getting that for free if you loosen this check to just `Vector128.IsHardwareAccelerated`.

Reply:
Ah yeah, this is much smarter, nice to know about wasm too.